src/main/java/io/vertx/core/net/impl/ConnectionBase.java

```diff
@@ -479,7 +482,10 @@ public String remoteName() {
   public SocketAddress remoteAddress() {
     SocketAddress address = remoteAddress;
     if (address == null) {
-      java.net.SocketAddress addr = chctx.channel().remoteAddress();
+      java.net.SocketAddress addr = chctx.channel().hasAttr(REMOTE_ADDRESS_OVERRIDE) ?
+        chctx.channel().attr(REMOTE_ADDRESS_OVERRIDE).getAndSet(null) :
```

Comment:
why `getAndSet` here instead of `get()` ? clearing the value will not remove the attribue

Reply:
Since we cache the address no need to keep the attribute around.

getAndRemove is deprecated with a note to use getAndSet with null value. 

https://netty.io/4.0/api/io/netty/util/Attribute.html